src/test_l2_hdmi_cec_driver.c

```diff
@@ -1,77 +1,427 @@
+    {
+        UT_LOG_ERROR("HdmiCecAddLogicalAddress failed with status: %d\n", status);
+        HdmiCecClose(handle);
+        return;
+    }
+
+    status = HdmiCecRemoveLogicalAddress(handle, logicalAddresses);
+    UT_LOG_DEBUG("Invoking HdmiCecRemoveLogicalAddress with handle: %d and logicalAddress: %d", handle, logicalAddresses);
+    UT_ASSERT_EQUAL(status, HDMI_CEC_IO_SUCCESS);
+    if (status != HDMI_CEC_IO_SUCCESS)
+    {
+        UT_LOG_ERROR("HdmiCecRemoveLogicalAddress failed with status: %d\n", status);
+        HdmiCecClose(handle);
+        return;
+    }
+
+    status = HdmiCecTx(handle, buf, len, &result);
+    UT_LOG_DEBUG("Invoking HdmiCecTx with handle: %d", handle);
+    UT_ASSERT_EQUAL(status, HDMI_CEC_IO_SUCCESS);
+    if (status != HDMI_CEC_IO_SUCCESS)
+    {
+        UT_LOG_ERROR("HdmiCecTx failed with status: %d\n", status);
+        HdmiCecClose(handle);
+        return;
+    }
+
+    UT_ASSERT_EQUAL(result, HDMI_CEC_IO_SENT_AND_ACKD);
+    if (result != HDMI_CEC_IO_SENT_AND_ACKD)
+    {
+        UT_LOG_ERROR("HdmiCecTx result: %d\n", result);
+    }
+
+    status = HdmiCecClose(handle);
+    UT_LOG_DEBUG("Invoking HdmiCecClose with handle: %d", handle);
+    UT_ASSERT_EQUAL_FATAL(status, HDMI_CEC_IO_SUCCESS);
+    if (status != HDMI_CEC_IO_SUCCESS)
+    {
+        UT_LOG_ERROR("HdmiCecClose failed with status: %d\n", status);
+    }
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+/**
+* @brief Test to verify the physical address of the HDMI CEC driver
+*
+* This test case verifies the physical address of the HDMI CEC driver by calling the HdmiCecGetPhysicalAddress() API. The test ensures that the physical address obtained is less than F.F.F.F. The test also includes error handling and assertions to check the return values of the APIs. The test ensures that the post-requisite API HdmiCecClose() is called in case of any test failure after the successful execution of the pre-requisite API HdmiCecOpen().
+*
+* **Test Group ID:** 02@n
+* **Test Case ID:** 005@n
+*
+* **Test Procedure:**
+* Refer to UT specification documentation [l2_hdmi_cec_driver_test_specification.md](../../docs/pages/l2_hdmi_cec_driver_test_specification.md)
+*/
+
+void test_l2_hdmi_cec_driver_VerifyPhysicalAddress(void)
+{
+    gTestID = 5;
+    int handle;
+    unsigned int physicalAddress;
+    HDMI_CEC_STATUS status;
+
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    // Step 1: Call the pre-requisite API HdmiCecOpen()
+    UT_LOG_DEBUG("Invoking HdmiCecOpen with valid handle");
+    status = HdmiCecOpen(&handle);
+    UT_ASSERT_EQUAL_FATAL(status, HDMI_CEC_IO_SUCCESS);
+
+    // Step 2: Call the API HdmiCecGetPhysicalAddress()
+    UT_LOG_DEBUG("Invoking HdmiCecGetPhysicalAddress with handle: %d", handle);
+    status = HdmiCecGetPhysicalAddress(handle, &physicalAddress);
+
+    // Step 3: Check the return status of HdmiCecGetPhysicalAddress()
+    if (status != HDMI_CEC_IO_SUCCESS)
```

Comment:
we need to remove 307 to 312
add UT_ASSERT_EQUAL(status, HDMI_CEC_IO_SUCCESS);

Reply:
We are using FATAL_ASSERT or return, only in the case of a failure, which would render the rest of the test as meaningless